Add MotherDuck tutorial (#17430)
cloudflare/cloudflare-docs: src/content/docs/pipelines/tutorials/query-data-with-motherduck/index.mdx

Create a new pipeline `clickstream-pipeline` using the [Wrangler CLI](/workers/wrangler/):

```sh
npx wrangler pipelines create clickstream-pipeline --r2 <BUCKET_NAME> --access-key-id <ACCESS_KEY_ID> --secret-access-key <SECRET_ACCESS_KEY>
```

Replace `<BUCKET_NAME>` with the name of your R2 bucket. Replace `<ACCESS_KEY_ID>` and `<SECRET_ACCESS_KEY>` with the keys you created in the previous step.

```output
🌀 Authorizing R2 bucket <BUCKET_NAME>
🌀 Creating pipeline named "clickstream-pipeline"
✅ Successfully created pipeline "clickstream-pipeline" with id <PIPELINE_ID>
🎉 You can now send data to your pipeline!
Example: curl "https://<PIPELINE_ID>.pipelines.cloudflare.com" -d '[{"foo": "bar"}]'
```

Make a note of the URL of your pipeline. You will need it in the next step.

## 2. Ingest data to R2

In this step, you will ingest data to your R2 bucket using `curl`. You will ingest the following JSON data to your R2 bucket:

<details>
<summary>

> Click to view the JSON data

Reviewer: ```suggestion
Select to view the JSON data
```

        "event_type": "form_submission",
        "form_id": "login-form",
        "page_url": "https://example.com/login",
        "timestamp": "2024-10-08T18:20:45.222Z"
      },
      {
        "event_id": "evt009",
        "event_type": "page_view",
        "page_url": "https://example.com/dashboard",
        "timestamp": "2024-10-08T18:20:50.333Z"
      },
      {
        "event_id": "evt010",
        "event_type": "feature_usage",
        "feature_id": "data_export",
        "page_url": "https://example.com/dashboard",
        "timestamp": "2024-10-08T18:22:30.444Z"
      }
    ],
    "device_info": {
      "device_type": "desktop",
      "operating_system": "macOS 10.15",
      "browser": "Chrome"
    },
    "referrer": "https://example.com/home"
  }
]
```
</details>

Run the following command to ingest the data to your R2 bucket using the pipeline you created in the previous step:

```sh
curl -X POST 'https://<PIPELINE_ID>.pipelines.cloudflare.com' -d '<JSON_DATA>'
```

Replace `<PIPELINE_ID>` with the ID of the pipeline you created in the previous step. Also, replace `<JSON_DATA>` with the JSON data provided above.

## 3. Connnect the R2 bucket to MotherDuck

> In this step, you will connect the R2 bucket to MotherDuck. You can connect the bucket to MotherDuck in several ways. You can learn about these different approaches in the [MotherDuck documentation](https://motherduck.com/docs/integrations/cloud-storage/cloudflare-r2/). In this tutorial, you will connect the bucket to MotherDuck using the MotherDuck dashboard.

Reviewer: ```suggestion
In this step, you will connect the R2 bucket to MotherDuck. You can connect the bucket to MotherDuck in several ways, which you can learn about from the [MotherDuck documentation](https://motherduck.com/docs/integrations/cloud-storage/cloudflare-r2/). In this tutorial, you will connect the bucket to MotherDuck using the MotherDuck dashboard.
```

> Login to the MotherDuck dashboard and click on your profile. Navigate to the **Secrets** page. Click on the **Add Secret** button and enter the following information:

Reviewer: ```suggestion
1. Log in to the MotherDuck dashboard and select your profile. 
2. Navigate to the **Secrets** page. 
3. Select on the **Add Secret** button and enter the following information:
```

- **Secret Name**: `Clickstream pipeline`
- **Secret Type**: `Cloudflare R2`
- **Access Key ID**: `ACCESS_KEY_ID` (replace with the Access Key ID you obtained in the previous step)
- **Secret Access Key**: `SECRET_ACCESS_KEY` (replace with the Secret Access Key you obtained in the previous step)

> Click on the **Add Secret** button to save the secret.

Reviewer: ```suggestion
Select the **Add Secret** button to save the secret.
```

## 4. Query the data

> In this step, you will query the data stored in the R2 bucket using MotherDuck. Navigate back to the MotherDuck dashboard and click on the **+** icon to add a new Notebook. Click on the **Add Cell** button to add a new cell to the notebook.

Reviewer: ```suggestion
In this step, you will query the data stored in the R2 bucket using MotherDuck. 

1. Navigate back to the MotherDuck dashboard and select the **+** icon to add a new Notebook. 
2. Select the **Add Cell** button to add a new cell to the notebook.
```

> In the cell, enter the following query and click on the **Run** button to execute the query:

Reviewer: ```suggestion
3. In the cell, enter the following query and select the **Run** button to execute the query:
```

```sql
SELECT * FROM `r2://<BUCKET_NAME>/<PATH_TO_FILE>`;
```

Replace the `<BUCKET_NAME>` placeholder with the name of the R2 bucket you created in the previous step. Replace the `<PATH_TO_FILE>` placeholder with the path to the file you uploaded in the previous step. You can find the path to the file by navigating to the object in the Cloudflare dashboard.

The query will return the data stored in the R2 bucket.

## Conclusion

> In this tutorial, you learned to create a pipeline and ingest data into a R2 bucket. You also learned how to connect the bucket with MotherDuck and query the data stored in the bucket. You can use this tutorial as a starting point to ingest data into an R2 bucket, and use MotherDuck to query the data stored in the bucket.

Reviewer: ```suggestion
In this tutorial, you learned to create a pipeline and ingest data into an R2 bucket. You also learned how to connect the bucket with MotherDuck and query the data stored in the bucket. You can use this tutorial as a starting point to ingest data into an R2 bucket and use MotherDuck to query the data stored in the bucket.
```
